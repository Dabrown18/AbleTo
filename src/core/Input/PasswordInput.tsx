import React, { ComponentProps, FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';
import { TextInput as PaperTextInput } from 'react-native-paper';

import TextInput from './TextInput';
import colors from '@src/core/colors';

type TextInputProps = Omit<ComponentProps<typeof TextInput>, 'error' | 'secureTextEntry'>;
export type Props = { label: string; error?: string; hideInput?: boolean; toggleShowPassword?(): void } & Omit<
    TextInputProps,
    'placeholder' | 'label'
>;

const PasswordInput: FunctionComponent<Props> = ({
    label,
    value,
    error,
    toggleShowPassword,
    hideInput = true,
    ...rest
}) => {
    const { t } = useTranslation();

    return (
        <TextInput
            label={label}
            value={value}
            mode="outlined"
            error={error}
            textContentType="password"
            secureTextEntry={hideInput}
            autoCapitalize="none"
            right={
                <PaperTextInput.Icon
                    color={colors.primary600}
                    name={hideInput ? 'eye' : 'eye-off'}
                    accessibilityLabel={t(
                        hideInput ? 'general.buttons.show_password' : 'general.buttons.hide_password'
                    )}
                    onPress={toggleShowPassword}
                />
            }
            {...rest}
        />
    );
};

export default PasswordInput;
