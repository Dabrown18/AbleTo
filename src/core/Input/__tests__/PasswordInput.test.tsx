import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';

import PasswordInput, { Props } from '../PasswordInput';

const defaultProps: Props = { label: 'Type your secrets in me!' };

const renderInput = (props: Props = defaultProps) => render(<PasswordInput {...props} />);

describe('PasswordInput snapshots', () => {
    test('Normal state', () => {
        const tree = renderInput().toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Error state', () => {
        const tree = renderInput({ ...defaultProps, error: 'Not good!' }).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Hidden state', () => {
        const tree = renderInput({ ...defaultProps, hideInput: true }).toJSON();

        expect(tree).toMatchSnapshot();
    });
});

describe('PasswordInput behaviour', () => {
    const text = 'password!';

    it('Handles user input', async () => {
        const onChangeText = jest.fn();
        const { getByLabelText } = renderInput({ ...defaultProps, onChangeText });

        const input = getByLabelText(defaultProps.label);
        fireEvent.changeText(input, text);

        expect(onChangeText).toHaveBeenCalled();
    });

    it('Calls password visibility toggle', async () => {
        const toggleShowPassword = jest.fn();

        const { getByLabelText } = renderInput({ ...defaultProps, toggleShowPassword });

        const visibilityToggle = getByLabelText('general.buttons.show_password');

        fireEvent.press(visibilityToggle);
        fireEvent.press(visibilityToggle);
        fireEvent.press(visibilityToggle);

        expect(toggleShowPassword).toHaveBeenCalledTimes(3);
    });

    it('Shows password input when visibility is on', async () => {
        const { getByLabelText, getByDisplayValue } = renderInput({ ...defaultProps, hideInput: false });

        const input = getByLabelText(defaultProps.label);
        fireEvent.changeText(input, text);

        const inputText = getByDisplayValue(text);
        expect(inputText).toBeTruthy();
    });

    it('Does not show password input when visibility is off', async () => {
        const { getByLabelText, queryByText } = renderInput({ ...defaultProps });

        const input = getByLabelText(defaultProps.label);
        fireEvent.changeText(input, text);

        const inputText = queryByText(text);
        expect(inputText).toBeFalsy();
    });
});
