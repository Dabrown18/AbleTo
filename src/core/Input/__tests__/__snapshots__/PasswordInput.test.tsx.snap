// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput snapshots Error state 1`] = `
Array [
  <View
    style={Object {}}
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#B00020",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
            },
            Object {
              "minHeight": 48,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <View
            collapsable={false}
            pointerEvents="none"
            style={
              Object {
                "backgroundColor": "#fff",
                "bottom": 4,
                "left": 10,
                "maxHeight": 2,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 12,
              }
            }
          />
          <Text
            collapsable={false}
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#fff",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "maxWidth": -28,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 29,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Type your secrets in me!
          </Text>
          <Text
            collapsable={false}
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#B00020",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 28,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Type your secrets in me!
          </Text>
          <Text
            collapsable={false}
            numberOfLines={1}
            style={
              Object {
                "color": "#B00020",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 1,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 28,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Type your secrets in me!
          </Text>
        </View>
        <TextInput
          accessibilityLabel="Type your secrets in me!"
          autoCapitalize="none"
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="#4B4845"
          secureTextEntry={true}
          selectionColor="#B00020"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              Object {
                "height": 40,
              },
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Object {
                "marginRight": 44,
                "paddingRight": 0,
              },
            ]
          }
          testID="text-input-outlined"
          textContentType="password"
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
            Object {
              "right": 12,
              "top": 15,
            },
          ]
        }
      >
        <View
          accessibilityLabel="general.buttons.show_password"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "height": 36,
                  "width": 36,
                },
                undefined,
                Object {
                  "margin": 0,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#85356F",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ó°ˆˆ
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <Text
    accessibilityLiveRegion="assertive"
    collapsable={false}
    onLayout={[Function]}
    style={
      Object {
        "color": "#B00020",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": Array [
          Object {
            "translateY": -0,
          },
        ],
        "writingDirection": "ltr",
      }
    }
  >
    Not good!
  </Text>,
]
`;

exports[`PasswordInput snapshots Hidden state 1`] = `
Array [
  <View
    style={Object {}}
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#E1E1E1",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
            },
            Object {
              "minHeight": 48,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <View
            collapsable={false}
            pointerEvents="none"
            style={
              Object {
                "backgroundColor": "#fff",
                "bottom": 4,
                "left": 10,
                "maxHeight": 2,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 12,
              }
            }
          />
          <Text
            collapsable={false}
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#fff",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "maxWidth": -28,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 29,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Type your secrets in me!
          </Text>
          <Text
            collapsable={false}
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#85356F",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 28,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Type your secrets in me!
          </Text>
          <Text
            collapsable={false}
            numberOfLines={1}
            style={
              Object {
                "color": "#4B4845",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 28,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Type your secrets in me!
          </Text>
        </View>
        <TextInput
          accessibilityLabel="Type your secrets in me!"
          autoCapitalize="none"
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="#4B4845"
          secureTextEntry={true}
          selectionColor="#85356F"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              Object {
                "height": 40,
              },
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Object {
                "marginRight": 44,
                "paddingRight": 0,
              },
            ]
          }
          testID="text-input-outlined"
          textContentType="password"
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
            Object {
              "right": 12,
              "top": 15,
            },
          ]
        }
      >
        <View
          accessibilityLabel="general.buttons.show_password"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "height": 36,
                  "width": 36,
                },
                undefined,
                Object {
                  "margin": 0,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#85356F",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ó°ˆˆ
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <Text
    accessibilityLiveRegion="assertive"
    collapsable={false}
    onLayout={[Function]}
    style={
      Object {
        "color": "#B00020",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "opacity": 0,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": Array [],
        "writingDirection": "ltr",
      }
    }
  />,
]
`;

exports[`PasswordInput snapshots Normal state 1`] = `
Array [
  <View
    style={Object {}}
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#E1E1E1",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
            },
            Object {
              "minHeight": 48,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <View
            collapsable={false}
            pointerEvents="none"
            style={
              Object {
                "backgroundColor": "#fff",
                "bottom": 4,
                "left": 10,
                "maxHeight": 2,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 12,
              }
            }
          />
          <Text
            collapsable={false}
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#fff",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "maxWidth": -28,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 29,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Type your secrets in me!
          </Text>
          <Text
            collapsable={false}
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#85356F",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 28,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Type your secrets in me!
          </Text>
          <Text
            collapsable={false}
            numberOfLines={1}
            style={
              Object {
                "color": "#4B4845",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 28,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Type your secrets in me!
          </Text>
        </View>
        <TextInput
          accessibilityLabel="Type your secrets in me!"
          autoCapitalize="none"
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="#4B4845"
          secureTextEntry={true}
          selectionColor="#85356F"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              Object {
                "height": 40,
              },
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Object {
                "marginRight": 44,
                "paddingRight": 0,
              },
            ]
          }
          testID="text-input-outlined"
          textContentType="password"
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 24,
              "justifyContent": "center",
              "position": "absolute",
              "width": 24,
            },
            Object {
              "right": 12,
              "top": 15,
            },
          ]
        }
      >
        <View
          accessibilityLabel="general.buttons.show_password"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                  "height": 36,
                  "width": 36,
                },
                undefined,
                Object {
                  "margin": 0,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#85356F",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ó°ˆˆ
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <Text
    accessibilityLiveRegion="assertive"
    collapsable={false}
    onLayout={[Function]}
    style={
      Object {
        "color": "#B00020",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "opacity": 0,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": Array [],
        "writingDirection": "ltr",
      }
    }
  />,
]
`;
